dynfu::kinect_fusion_opencl_update_reconstruction_pipeline_block:

jordan-heemskerk opened the pull request:
- First run at implementation
- Close #66

RobertLeahy commented on include/dynfu/kinect_fusion_opencl_update_reconstruction_pipeline_block.hpp:
```diff
@@ -0,0 +1,89 @@
+	
+	
+	/**
+	 *	
```
Shouldn't there be something here?

RobertLeahy commented on include/dynfu/update_reconstruction_pipeline_block.hpp:
```diff
@@ -6,6 +6,12 @@
 
 
+#include <dynfu/pipeline_value.hpp>
+#include <dynfu/depth_device.hpp>
```
Not alphabetical

RobertLeahy commented on include/dynfu/update_reconstruction_pipeline_block.hpp:
```diff
@@ -36,7 +63,36 @@ namespace dynfu {
+			 *
+			 *	\return
+			 *		A \ref pipeline_value representing the TSDF generated from \em frame.
+			 *	
```
I wouldn't have this empty line here but that doesn't **really** matter...

RobertLeahy commented on src/test/kinect_fusion_opencl_update_reconstruction_pipeline_block.cpp:
```diff
@@ -0,0 +1,525 @@
+			
+			std::size_t i(0);
+			
+			/*
```
Is this commented out code important?  I'd prefer not to commit a bunch of commented out code...

jordan-heemskerk replied:
Could put it in a function and then never call it...?

RobertLeahy replied:
But you'd want the function to be `static` and then GCC would complain about a `static` function that's never called.

You could guard it with the preprocessor?  Try `#if 0` ... `#endif`?

jordan-heemskerk committed:
dynfu::kinect_fusion_opencl_update_reconstruction_pipeline_block:

 - This has not been verified to produce truly correct TSDF
 - It produces something that looks like a TSDF
 - Unit test to ensure no exceptions are thrown when it is run
 - Closes #66

RobertLeahy commented on include/dynfu/update_reconstruction_pipeline_block.hpp:
```diff
@@ -23,6 +28,27 @@ namespace dynfu {
+			
+		
+		
+			class value_type {
```
Maybe this should be documented.

RobertLeahy commented on include/dynfu/update_reconstruction_pipeline_block.hpp:
```diff
@@ -36,7 +62,35 @@ namespace dynfu {
+			 *	\return
+			 *		A \ref pipeline_value representing the TSDF generated from \em frame.
+			 */
+			virtual value_type operator ()(depth_device::value_type::element_type & frame, std::size_t width, std::size_t height, Eigen::Matrix3f k, Eigen::Matrix4f T_g_k, value_type v=value_type{}) = 0;
```
I usually put a space between `operator ()` and the beginning of the arguments, but that's a style nitpicky thing that probably doesn't matter...

RobertLeahy commented on src/kinect_fusion_opencl_update_reconstruction_pipeline_block.cpp:
```diff
@@ -0,0 +1,129 @@
+			tsdf_width_(tsdf_width),
+			tsdf_height_(tsdf_height),
+			tsdf_depth_(tsdf_depth)
+
```
Why is this empty line here?

RobertLeahy commented on src/kinect_fusion_opencl_update_reconstruction_pipeline_block.cpp:
```diff
@@ -0,0 +1,129 @@
+		
+		if (k_ != k) {
+		
+			k_ = std::move(k);
```
Add `#include <utility>` at the top of the file.

RobertLeahy commented on src/kinect_fusion_opencl_update_reconstruction_pipeline_block.cpp:
```diff
@@ -0,0 +1,129 @@
+			t_g_k_ = std::move(t_g_k);
+			
+			// generate the projection matrix
+			Eigen::Matrix4f proj = Eigen::Matrix4f::Zero();
```
I'd add `#include <Eigen/Dense>` but that's more of a preference than a rule.

RobertLeahy commented on src/test/kinect_fusion_opencl_update_reconstruction_pipeline_block.cpp:
```diff
@@ -0,0 +1,128 @@
+					 0.0f, -0.585f, 16.0f,
+					 0.0f, 0.0f, 1.0f;
+					 
+				T = Eigen::Matrix4f::Identity();
```
The fact that this is uppercase **and** named `T` (which usually means a type/template parameter) is jarring.

RobertLeahy commented on src/test/kinect_fusion_opencl_update_reconstruction_pipeline_block.cpp:
```diff
@@ -0,0 +1,128 @@
+		dynfu::cpu_pipeline_value<std::vector<float>> pv;
+		std::vector<float> v{0.0f, 5.0f, 10.0f, 15.0f, 13.0f, 40.0f, 12.0f, 10.0f, 8.0f,19.0f,202.0f,102.0f,84.0f,293.0f,292.0f,293.0f};
+
+		pv.emplace(std::move(v));
```
`#include <utility>`

RobertLeahy commented on src/test/kinect_fusion_opencl_update_reconstruction_pipeline_block.cpp:
```diff
@@ -0,0 +1,128 @@
+				std::size_t i(0);
+				for (auto && t_ : ts) {
+					
+					std::cout << t_ <<",";
```
`#include <iostream>`

RobertLeahy commented on src/kinect_fusion_opencl_update_reconstruction_pipeline_block.cpp:
```diff
@@ -0,0 +1,129 @@
+		tsdf_kernel_.set_arg(11, std::uint32_t(frame_height));
+		
+		// Ready to run the kernel
+		std::size_t tsdf_extent[] = {tsdf_width_, tsdf_height_, tsdf_depth_};
```
`#include <cstddef>`
